functions-eventbrite.php

```diff
@@ -1,121 +1,220 @@
-			$title   = $obj->events[ $i ]->name->text;
-			$image   = $obj->events[ $i ]->logo->url;
-			$date    = date( 'l j F Y, H:i', strtotime( $obj->events[ $i ]->start->local ) );
-			$status  = $obj->events[ $i ]->ticket_classes;
-			$tickets = tna_ebapi_event_status( $status );
-			$online  = tna_ebapi_event_online( $obj->events[ $i ]->online_event );
+	/**
+	 * @param $title
+	 * @param $url
+	 * @param $i
+	 *
+	 * @return string
+	 */
+	public function gtm( $title, $url, $i ) {
 
-			$html .= '<li>';
+		$gtm = 'data-gtm-name="%s" data-gtm-id="%s" data-gtm-position="event_position_%s" data-gtm-creative="event_page_eventbrite"';
 
-			$html .= '<div class="event-img">' . $online;
+		return sprintf( $gtm, $title, $url, $i );
+	}
 
-			$html .= '<a href="' . $url . '" target="_blank"><img src="' . $image . '" alt="' . $title . '"></a></div>';
+	/**
+	 * @param $online
+	 *
+	 * @return string
+	 */
+	public function event_online( $online ) {
 
-			$html .= '<div class="event-text">';
+		$html = '';
 
-			$html .= '<p><span class="event-date">' . $date . '</span></p>';
+		if ( $online == true ) {
+			$html = '<div class="online-event"><span>Online event</span></div>';
+		}
 
-			$html .= '<h4><a href="' . $url . '" target="_blank">' . $title . '</a></h4>';
+		return $html;
+	}
 
-			$html .= '<p class="event-status">' . $tickets . '</p>';
+	/**
+	 * @param $online
+	 * @param $url
+	 * @param $image
+	 * @param $title
+	 * @param $i
+	 *
+	 * @return string
+	 */
+	public function event_image( $online, $url, $image, $title, $i ) {
 
-			$html .= '</div></li>';
+		$gtm = $this->gtm( $title, $url, $i );
 
-		}
-	} else {
+		$html = '<div class="event-img">%s<a href="%s" %s target="_blank"><img src="%s" alt="%s"></a></div>';
 
-		$html .= '<h2>No events found</h2>';
+		return sprintf( $html, $online, $url, $gtm, $image, $title );
+	}
+
+	/**
+	 * @param $date
+	 * @param $url
+	 * @param $title
+	 * @param $tickets
+	 * @param $i
+	 *
+	 * @return string
+	 */
+	public function event_text( $date, $url, $title, $tickets, $i ) {
+
+		$gtm = $this->gtm( $title, $url, $i );
 
+		$html = '<div class="event-text"><p>%s</p><h4><a href="%s" %s target="_blank">%s</a></h4><p class="event-status">%s</p></div>';
+
+		return sprintf( $html, $date, $url, $gtm, $title, $tickets );
 	}
 
-	$html .= '</ul></div>';
+	/**
+	 * @param $organiser
+	 * @param $category
+	 * @param $token
+	 * @param $number
+	 *
+	 * @return string
+	 */
+	public function display( $organiser, $category, $token, $number ) {
 
-	return $html;
-}
+		$url  = $this->url( $organiser, $category, $token );
+		$json = $this->get_json( $url );
 
-function tna_ebapi_event_status( $status ) {
+		$html = '<div id="tna_ebapi_events" class="track-outbound"><ul class="tna-event-list">';
 
-	$tickets = '';
+		if ( $json ) {
 
-	if ($status) {
+			$obj = json_decode( $json );
 
-		$count = count($status);
+			if ( $obj->status_code == 400 || $obj->status_code == 401 ) {
```

Comment:
Undefined property on line 177, please check.
Notice: Undefined property: stdClass::$status_code in /Applications/MAMP/htdocs/sites/tna-base/wp-content/plugins/tna-eventbrite-api/functions-eventbrite.php on line 177